group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/AssignmentSpec.java

```diff
@@ -28,12 +28,12 @@ public class AssignmentSpec {
     /**
      * The members keyed by member id.
      */
-    final Map<String, AssignmentMemberSpec> members;
+    private final Map<String, AssignmentMemberSpec> members;
 
     /**
      * The topics' metadata keyed by topic id
      */
-    final Map<Uuid, AssignmentTopicMetadata> topics;
+    private final Map<Uuid, AssignmentTopicMetadata> topics;
```

Comment:
Did you mean to add the partitions with their rack ids to `AssignmentTopicMetadata`? What's your plan?

Reply:
I didn't think about the full implementation yet and where the rackIds will be passed, figured we could edit it when it came to implementing the algorithm. The topicIdToPartition data structure could be populated using this so maybe we can add a field for a map of partition to its rackIds